src/averageAges.js

```diff
@@ -37,7 +40,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(({ sex, name }) =>
+      sex === 'f' && people.some(({ mother }) => name === mother))
+    : people.filter(({ sex }) => sex === 'f');
+
+  return women.reduce(
+    (prev, { born, died }) => prev + (died - born), 0)
```

Comment:
Actually it is good, but u can use variable for died - born formula or alternatively for 1st and 2nd tasks you can do a function to calculate years
```
function calculateAge(array) {
  return array.reduce((acc, person) => {
    const age = person.died - person.born;
    return acc + age;
  }, 0) / array.length;
}
```